python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -130,9 +131,14 @@ def runtimeEvalString(this_obj, attr_name, val):
 
     return new_val
 
-
+_stored_evals = {}
 
 def raw_eval(val):
+    if val not in _stored_evals:
+        _stored_evals[val] = _really_eval(val)
+    return copy.deepcopy(_stored_evals[val])
+
+def _really_eval(val):
```

Comment:
Sharing the decorators makes profiling more difficult, abstracting things helps me introduce another layer of logic rather than adding pointless indents. I can change the name but I think another function is called for here.